src/extractor/guidance/intersection_handler.cpp

```diff
@@ -383,136 +393,140 @@ std::size_t IntersectionHandler::findObviousTurn(const EdgeID via_edge,
+            in_way.name_id, out_node.name_id, name_table, street_name_suffix_table);
+        if (out.same_name_id)
+            ++continue_count;
 
-        if (same_name && (best_continue == 0 || (continue_class.GetPriority() >
-                                                     out_data.road_classification.GetPriority() &&
-                                                 in_classification != continue_class) ||
-                          (deviation < best_continue_deviation &&
-                           out_data.road_classification == continue_class) ||
-                          (continue_class != in_classification &&
-                           out_data.road_classification == continue_class)))
+        if (!out.road->entry_allowed)
         {
-            best_continue_deviation = deviation;
-            best_continue = i;
+            ++continue_count_valid;
+            continue;
         }
 
-        const auto current_best_class =
-            node_based_graph.GetEdgeData(intersection[best_continue].eid).road_classification;
+        out.index = i;
+        out.deviation_from_straight = angularDeviation(intersection[i].angle, STRAIGHT_ANGLE);
+        out.same_classification = out_node.road_classification == in_way.road_classification;
+        out.same_or_higher_priority =
+            out_node.road_classification.GetPriority() <= in_way.road_classification.GetPriority();
 
-        // don't prefer low priority classes
-        if (best != 0 && out_data.road_classification.IsLowPriorityRoadClass() &&
-            !current_best_class.IsLowPriorityRoadClass())
-            continue;
+        out_ways.push_back(out);
+    }
+    const std::pair<std::int64_t, std::int64_t> num_continue_names =
+        std::make_pair(continue_count, continue_count_valid);
 
-        const bool is_better_choice_by_priority =
-            best == 0 || obviousByRoadClass(in_data.road_classification,
-                                            out_data.road_classification,
-                                            current_best_class);
+    const auto End = end(out_ways);
+    // dead end, or otherwise no exitable ways
+    if (out_ways.size() == 0)
+        return 0;
 
-        const bool other_is_better_choice_by_priority =
-            best != 0 && obviousByRoadClass(in_data.road_classification,
-                                            current_best_class,
-                                            out_data.road_classification);
+    // sort out ways by what they share with the in way
+    const auto sort_by = [](const out_way &lhs, const out_way &rhs) {
+        auto left_tie = std::tie(
+            rhs.deviation_from_straight, lhs.same_classification, lhs.same_or_higher_priority);
+        auto right_tie = std::tie(
+            lhs.deviation_from_straight, rhs.same_classification, rhs.same_or_higher_priority);
+        return left_tie > right_tie;
+    };
+    std::stable_sort(begin(out_ways), End, sort_by);
```

Comment:
by doing stable sort, you prefer right turns over left turns in the following `find_if` calls. Intended? Can this have side-effects?